gems/sorbet-runtime/test/types/types.rb

```diff
@@ -442,6 +442,24 @@ def subtype_of?(other)
         allocs_when_invalid = counting_allocations {type.valid?(invalid)}
         assert_equal(0, allocs_when_invalid)
       end
+
+      it 'gives the right error when passed an unrelated enumerable' do
```

Comment:
If I understand this right, it's not possible to reproduce this anymore given the recent changes to remove runtime type checking of collection types, right? e.g. #3293

Reply:
that is, to write a test you had to call `error_message_for_obj` directly, but I don't think there's a user-visible way to trigger the behavior in #2808 anymore.